src/rgw/driver/rados/rgw_d3n_datacache.cc

```diff
@@ -40,9 +40,10 @@ int D3nCacheAioWriteRequest::d3n_libaio_prepare_write_op(bufferlist& bl, unsigne
     return r;
   }
   r = 0;
-
+#if _POSIX_C_SOURCE >= 200112L
```

Comment:
Can you add a comment detailing why needed for Apple?
(for other systems, I do not suppose we need to care for pre-2001 'C')

Reply:
macOs stdlib doesn't implement this posix function. According to https://man7.org/linux/man-pages/man2/posix_fadvise.2.html the correct way to test for it is 

>Feature Test Macro Requirements for glibc (see
   [feature_test_macros(7)](https://man7.org/linux/man-pages/man7/feature_test_macros.7.html)):

       posix_fadvise():
           _POSIX_C_SOURCE >= 200112L